### ISSUE TITLE
Change Windows Default Terminal to Powershell

### ISSUE THREAD

**Issue by tiny-pangolin** [Ylianst/MeshCentral#1142]
I love having the full shell available in meshcentral, however it is still using cmd instead of powershell I would like to see this since Powershell is more full featured and installed by default in every supported addition of Windows

**Comment by krayon007** [comment]
If we were to do the change, I would only have it default to powershell if ConPTY is detected, because older versions of windows don't have proper pseudo console support, which makes powershell's UI not as stable as ConPTY.

**Comment by tiny6996** [comment]
would this be something that needed to be done on the mesh server or be in the client

**Comment by krayon007** [comment]
no, this can totally all be done client side. It's actually a pretty minor change in meshcore.js.
I'm guessing @Ylianst  will probably want to make this a configurable setting in the server, to tell the agent which is the preferred/default.

**Comment by tiny6996** [comment]
Mind if I take a swing at I want to start contributing to meshcentral

Apr 11, 2020 16:37:04 Bryan Roe <notifications@github.com>:

**Comment by krayon007** [comment]
Go for it :)

**Comment by Ylianst** [comment]
Just to make sure this is known, if you right click on the "Connect" button in the terminal, you can select "PowerShell or have cmd or powershell run as current user. Let me know if this is ok? It is available right now.

![x](https://user-images.githubusercontent.com/1319013/79058254-714eaa80-7c20-11ea-85da-6307a900de11.png)

**Comment by tiny6996** [comment]
I was not aware of this. Can I still submit a patch to make powershell the default?

Apr 11, 2020 20:17:06 Ylian Saint-Hilaire <notifications@github.com>:

**Comment by Ylianst** [comment]
If that is ok, I will change the user interface to make it customizable. This way, everyone will be happy. I rather not change the default if possible.

**Comment by tiny6996** [comment]
That this fine thanks for your rapid response

Apr 11, 2020 20:21:57 Ylian Saint-Hilaire <notifications@github.com>:

**Comment by si458** [comment]
just looking at this,
would you want to change the default terminal connect action to be for the whole server, per domain, per group, per device, per user?
working from left to right from that list is basically more work!
whole server should be easy, but per user would be more difficult
